src/cpp/rtps/xmlparser/XMLDynamicParser.cpp

```diff
@@ -441,7 +457,10 @@ XMLP_ret XMLParser::parseXMLBitsetDynamicType(
         }
     }
 
-    XMLProfileManager::insertDynamicTypeByName(name, typeBuilder);
+    if (false == XMLProfileManager::insertDynamicTypeByName(name, typeBuilder))
+    {
+        types::DynamicTypeBuilderFactory::get_instance()->delete_builder(typeBuilder);
+    }
```

Comment:
Also in this case, shell we return an error if `XMLProfileManager::insertDynamicTypeByName` return false ?
```suggestion
        return XMLP_ret::XML_ERROR;
        }
```